File: contests/abc051/abc051_b.py
# B - Sum of Three Integers
# https://atcoder.jp/contests/abc051/tasks/abc051_b

K, S = map(int, input().split())

count = 0
for X in range(K+1):
    for Y in range(K+1):
        if 0 <= (S-X-Y) <= K:
            count += 1

print(count)

File: contests/abc051/abc051_b.cpp
// B - Sum of Three Integers
// https://atcoder.jp/contests/abc051/tasks/abc051_b

#include <iostream>
using namespace std;

int main() {
    int k, s; cin >> k >> s;
    int count = 0;
    for(int x=0;x<=k;x++) {
        for(int y=0;y<=k;y++) {
            if(s-x-y>=0 && s-x-y<=k) count++;
        }
    }
    cout << count << endl;
}
